### ISSUE TITLE
Work with tail -f

### ISSUE THREAD

**Issue by SimenB** [fiatjaf/jiq#1]
Would it be possible for this to work on stdin as well?

I'd like to tail a log and send it into this, and be able to interactively filter it. `jq` handles stdin just fine, if that matters 😄

**Comment by fiatjaf** [comment]
Well, it does work with stdin content. Actually, it only works with stdin, I think. It should work with other types of input.

`echo {} | jiq` actually raises an error, but it shouldn't. However, `echo {"a": "b"} | jiq` and other valid JSON over stdin works well. What error are you seeing?

**Comment by SimenB** [comment]
Yeah, that was imprecise of me, sorry.
What I meant is e.g. `tail`ing a log using the `follow` option. If I do `tail -f something.json | jq`, `jq` will also format anything new added to the `something.json` file. If I pipe into `jiq` though, I never get anything at all. Not even the original content of the file.

My real use case is that I currently tail a log and send it into `jq` for formatting, which works fine, but I'd also like to be able to apply filters on the fly, instead of restarting jq all the time.

Does that make sense?

**Comment by fiatjaf** [comment]
Oh, I see it now. This seems complicated. I don't know how `tail -f` works, nor how `jq` does what it does. It doesn't seem to work if you modify an existing JSON object on the file, only if you add a new line, right? 

Sorry, but I think I will not be able to include this feature anytime soon, but if you (or anyone else) did a PR I would be glad to merge it.

**Comment by SimenB** [comment]
Yes, just adding a new line. And logs are append only, so no real issue.

Ok, that's fine 😄

**Comment by fiatjaf** [comment]
It has something to do with reading Stdin as a stream, right? It shouldn't be difficult.

**Comment by SimenB** [comment]
Yup, sounds right.
I don't know Go at all, so won't be able to provide a PR, unfortunately.
